mlflow/metrics/metric_definitions.py

```diff
@@ -17,17 +17,24 @@ def standard_aggregations(scores):
     }
 
 
-def _validate_text_data(data, metric_name, column_name):
-    """Validates that the data is text and is non-empty"""
-    if len(data) == 0:
+def _validate_text_data(data, metric_name, column_name=None):
+    """Validates that the data is a list of strs and is non-empty"""
+    if data is None or len(data) == 0:
         return False
 
     for row, line in enumerate(data):
         if not isinstance(line, str):
-            _logger.warning(
-                f"Cannot calculate {metric_name} for non-string inputs. "
-                + f"Non-string found for {column_name} on row {row}. skipping metric logging."
-            )
+            if column_name:
+                _logger.warning(
+                    f"Cannot calculate {metric_name} for non-string inputs. "
+                    f"Non-string found for {column_name} on row {row}. skipping metric logging."
+                )
+            else:
+                _logger.warning(
+                    f"Cannot calculate {metric_name} for non-string inputs. "
+                    f"Non-string found for the column specified by `predictions` "
+                    f"or the model output column on row {row}. Skipping metric logging."
+                )
```

Comment:
Rather than special casing it this way. Can we just create another function `_validate_prediction_data_as_str` or something?